### ISSUE TITLE
[IP #090] Integrate prototype for Eclipse Extension Points

### ISSUE THREAD

**Issue by RomanLangrehr** [Beagle-PSE/Beagle#576]
Integrate the prototype for Eclipse Extension Points into Beagle.

Depends on: IP#020

**Comment by RomanLangrehr** [comment]
Implementation is done.

**Comment by jGleitz** [comment]
This is very hard to test. Automated tests would require a setup we don’t have yet. I propose we don’t expect to have it tested until Thursday.

**Comment by jGleitz** [comment]
I’ll test this.

**Comment by RomanLangrehr** [comment]
:+1:

**Comment by jGleitz** [comment]
@RomanLangrehr I made [a bunch of changes (diff)](https://github.com/Beagle-PSE/Beagle/compare/9dc97378781ccd42bd22aaa010bab89e598d45b1...ip090-extension-points) to the code, please have a look at it and verify they’re okay.

**Comment by RomanLangrehr** [comment]
Changes lgtm.

**Comment by jGleitz** [comment]
I propose to leave this untested. That’s because:

 * Any “simple” test will mainly test the Eclipse API which does not make sense.
 * A more advanced test would mock Eclipse’s Platform class and everything it returns. This is however pretty difficult and I was not able to do so (yes, I tried with Power Mockito).
 * The classes *will*  automatically be tested in integration testing.
 * The classes are very simple.

In short: The effort to test this is big, the potential use is small. Maybe reviewing this carefully is enough.

**Comment by jGleitz** [comment]
If someone comes up with a clever way to test this I’d highly appreciate it and would happily implement it, though.
